crates/wallet/src/types.rs

```diff
@@ -20,6 +20,7 @@ use serde::{Deserialize, Serialize};
 
 /// Types of keychains
 #[derive(Serialize, Deserialize, Debug, Clone, Copy, PartialEq, Eq, Hash, Ord, PartialOrd)]
+#[serde(tag = "type")]
```

Comment:
For some reason, this makes `bdk_file_store` fail. Looking into it.

Reply:
Okay, so [`bincode` encodes all enums as `u32`](https://github.com/bincode-org/bincode?tab=readme-ov-file#why-does-bincode-not-respect-repru8)...

Reply:
However, removing this breaks `bdk_sqlite_store` with some keychains/anchor types due to the following:

1. How the `json_extract()` method works in sqlite: https://sqlite.org/json1.html#jexb
2. We are storing keychains/anchors as JSONB

The `json_extract()` method may not always return JSON values, and may fail parsing by `serde_json`.

Reply:
We should just use `json()` instead of `json_extract()`.

Reply:
Oh ya I should have documented that I ran into this issue with the non-struct `KeychainKind` enum. I'll try using `json()`, I think I tried it already but maybe not. Worst case we could store `keychain` as a simple json string for maximum compatibility, it just feels less cool.

Reply:
Fixed here: https://github.com/notmandatory/bdk/pull/7/commits/06565995723b2c8bfe0d2e9c3d51b19e6b9619da

Sorry the commit mixed other things in as well.

Reply:
Using JSONB seems to be a good idea. Fixed in the commit above. I tend to try things out while I review, it feels more thorough for me.